CMakeLists.txt

```diff
@@ -23,6 +23,10 @@ project(ecdaa
 
 add_compile_options(-Werror -Wall -Wextra -std=c99)
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        SET(CMAKE_C_FLAGS "-g -O0 -fprofile-arcs -ftest-coverage")
```

Comment:
Should these be added to a new build type, so that the debug build doesn't always produce the gcov files?  E.g., `DebugWithCoverage`.

See https://cmake.org/Wiki/CMake_FAQ#How_can_I_extend_the_build_modes_with_a_custom_made_one_.3F and https://stackoverflow.com/questions/11437692/how-to-add-a-custom-build-type-to-cmake-targeting-make

Reply:
My though here is that developers may run the debug build locally, but not want the code coverage overhead and output.

Reply:
The build-type will now be `DebugWithCoverage`, and just `Debug` is left alone. Good call.